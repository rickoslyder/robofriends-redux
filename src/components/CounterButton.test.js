import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import CounterButton from './CounterButton';

const mockButtonComponent = <CounterButton color={'red'}/>

test('expect to render CounterButton component', () => {
    render(mockButtonComponent)
    const buttons = screen.getAllByRole("button")
    expect(buttons).toBeDefined();
    expect(buttons.length).toEqual(1)
})

test('expect mock CounterButton component JSX to match snapshot', () => {
    expect(mockButtonComponent).toMatchSnapshot()
})

test('expect rendered mock CounterButton component to match snapshot', () => {
    expect(render(mockButtonComponent).baseElement).toMatchSnapshot()
})

test('expect click to increment count by 1', () => {
    render(mockButtonComponent)
    const button = screen.getByRole("button")
    fireEvent.click(button)
    expect(button).toHaveTextContent("Count: 1")
    fireEvent.click(button)
    expect(button).toHaveTextContent("Count: 2")
})