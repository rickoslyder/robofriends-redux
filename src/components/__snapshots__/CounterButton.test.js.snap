// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect mock CounterButton component JSX to match snapshot 1`] = `
<CounterButton
  color="red"
/>
`;

exports[`expect rendered mock CounterButton component to match snapshot 1`] = `
<body>
  <div>
    <button
      color="red"
    >
      Count: 
      0
    </button>
  </div>
</body>
`;
